GVFS/FastFetch/NativeUnixMethods.cs

```diff
@@ -0,0 +1,146 @@
+﻿using GVFS.Common.Tracing;
+using System;
+using System.ComponentModel;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.InteropServices;
+
+namespace FastFetch
+{
+    public class NativeUnixMethods 
+    {
+        public const int ReadOnly = 0x0000;
+        public const int WriteOnly = 0x0001;
+
+        public const int Create = 0x0200;
+        public const int Truncate = 0x0400;
+        private const int InvalidFileDescriptor = -1;
+
+        public static unsafe void WriteFile(ITracer tracer, byte* originalData, long originalSize, string destination, ushort mode)
+        {
+            int fileDescriptor = InvalidFileDescriptor;
+            try
+            {
+                fileDescriptor = Open(destination, WriteOnly | Create | Truncate, mode);
+                if (fileDescriptor != InvalidFileDescriptor)
+                {
+                    IntPtr result = Write(fileDescriptor, originalData, (IntPtr)originalSize);
+                    if (result.ToInt32() == -1)
+                    {
+                        throw new Win32Exception(Marshal.GetLastWin32Error());
+                    }
+                }
+            }
+            catch (Win32Exception e)
+            {
+                tracer.RelatedError("Error writing file {0}. Win32Exception: {1}", destination, e);

```

Comment:
- Put `destination` and `exception` in an `EventMetadata`
- Call `e.ToString()` so that we have the exception call stack